webapp/views/launch/script.js

```diff
@@ -470,37 +470,45 @@ function buttonElement(action) {
 }
 
 function relationSelect() {
-  const relationOption = ['AND', 'OR', 'NOT'].map(el => `<option value="${el}">${el}</option>`);
+  const relationOption = ['AND', 'OR'].map(el => `<option value="${el}">${el}</option>`);
   return $(`<select class="rel custom-select mb-2">${relationOption}</select>`)
 }
 
 function conditionElement(data, id) {
-  const container = $(`<div class='d-flex mb-2' id='${id}'></div>`)
+  const container = $(`<div class='card-body border-info m-0 p-10'></div>`)
+  const row1 = $(`<div class='d-flex mb-2' id='${id}'><small>query attribute</small></div>`)
   const columns = data.map(item => `<option value="${item.column}">${item.column}</option>`)
-  const select = $(`<select class='custom-select mr-2 sel'>${columns}</select>`)
+  const select = $(`<select class='custom-select mr-2 ml-2 sel'>${columns}</select>`)
   const remove = $('<button type="button" class="btn btn-secondary ml-2">&times;</button>')
-  const info = $(`<div class="mb-2 text-light bg-dark">${'min: ' + data[0].bounds[0] + ' max: ' + data[0].bounds[1]}</div>`)
-  const inputs = $(`
+  // const info = $(`<div class="mb-2 text-light bg-dark">${'min: ' + data[0].bounds[0] + ' max: ' + data[0].bounds[1]}</div>`)
+  const row2 = $(`
   <div class='d-flex justify-content-between mb-2'>
-    <label for='${id}-input-min'>min</label>
+    <small>min <span class='min-badge badge badge-secondary'> >= ${data[0].bounds[0]}</span></small>
```

Comment:
Is it possible that `data` is undefined, or length of `data` is 0, or `data[0]` is undefined, or `data[0].bounds` is undefined or zero-length? If yes, it has to be dealt with.

Reply:
If the max or min is not given by GRASS (the `v.db.univar` command), then it is a data problem, which should be handled on the node side. So far I haven't seen such exceptions, so the `v.db.univar` might already handles this for us. Let's deal with this when we run into one exception.

Reply:
OK